Calibration minigame

PiersSK opened the pull request:
Adds to the bridge terminal, a second temporary screen which is used in the weapons phase task 'calibrate weapons'.

Before task:
- shows locked status screen
Start of Phase 2:
- status screen shows that a target has been authorised
After Warmup Task:
- game is playable

- Game allows you to scroll an arial map (temp). When you hover over a location it'll tell you it's name. Red chevrons at the side of the screen show you the direction of the authorised target if it's off screen.
- When you hover over the authorised target, you can press E to begin the calibration. 
- Calibration launches instantly, and has the player match a constantly moving ball to fill a bar
    - If the bar goes to 0, the terminal is locked for 15 seconds
    - If the bar fills, the task is completed and a locked/success screen shows on the terminal

PiersSK committed:
feat(code): adds basic calibration map UI

PiersSK committed:
feat(code): adds lock in terminal for calibration

PiersSK committed:
feat(code): tidies button prompt code for expansion

PiersSK committed:
feat(code): adds scroll map button prompt

PiersSK committed:
feat(code): improves stage one UI

- sfx
- moved location of info
- color changing to highlight location found
- crosshair snap to location

PiersSK committed:
feat(code): adds basic calibration minigame

PiersSK committed:
fix(code): improves mouse tracking

PiersSK committed:
feat(code): adds controller support probably

PiersSK committed:
feat(code): add minigame start and fail

PiersSK committed:
feat(code): adds full task flow with success

PiersSK committed:
fix(code): game releases player on fail

PiersSK committed:
feat(code): removes old level functionality

PiersSK committed:
feat(code): adds ascending beep to minigame

PiersSK committed:
fix(code): stops buttons prompts overlapping

PiersSK committed:
fix(code): hides temp console buttons

OrcaOfTime reviewed (commented):
Some notes on an initial playthrough:

The dynamic button prompts weren't updating depending on the control scheme I was using. X was the default interact button whilst I was expecting E as I was using keyboard. 
Also ctrl for exiting terminals, which instead should be esc.

Calibration game had a fun feel, the default targeting speed felt a bit sluggish. 
Not sure if intended behaviour but when I followed the red arrows to other targets, then moved away from those targets until they were off screen, the red arrows did not appear again, I did not interact with them.

The mini-game was fun, could we curve the bar we control if it is following a circular path.

PiersSK committed:
fix(code): corrects button prompts

PiersSK committed:
feat(code): improves minigame marker

- block becomes a circle arc
- parameterises match angle
- increases map scroll speed by 25%
